spot_utils/structures/image.py

```diff
@@ -0,0 +1,57 @@
+"""Data structures for images and pointclouds."""
+
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy as np
+from bosdyn.client.math_helpers import SE3Pose
+
+
+@dataclass
+class Intrinsics:
+    """Camera intrinsics."""
+
+    rows: int
+    cols: int
+    fx: float
+    fy: float
+    cx: float
+    cy: float
+
+
+@dataclass
+class RGBImage:
+    """Color image."""
+
+    rgb: np.ndarray
+    frame: SE3Pose
+    intrinsics: Intrinsics
+
+
+@dataclass
+class DepthImage:
+    """Depth image."""
+
+    depth: np.ndarray
```

Comment:
general comment: instead of `np.ndarray` for type annotations, let's use `NDArray[...]` from `numpy.typing`. for example, this might be `NDArray[np.uint8]`